mmdet/datasets/builder.py

```diff
@@ -103,41 +106,73 @@ def build_dataloader(dataset,
         dist (bool): Distributed training/test or not. Default: True.
         shuffle (bool): Whether to shuffle the data at every epoch.
             Default: True.
+        runner (dict): The config dictionary of runner.
         kwargs: any keyword argument to be used to initialize DataLoader
 
     Returns:
         DataLoader: A PyTorch dataloader.
     """
     rank, world_size = get_dist_info()
-    if dist:
-        # DistributedGroupSampler will definitely shuffle the data to satisfy
-        # that images on each GPU are in the same group
-        if shuffle:
-            sampler = DistributedGroupSampler(
-                dataset, samples_per_gpu, world_size, rank, seed=seed)
+
+    runner_type = 'EpochBasedRunner' if runner is None else runner['type']
```

Comment:
If you only need `runner_type`, then why pass in the entire `runner` dictionary?